### PR TITLE
improvements

### PR DESCRIPTION
**Pull request by PedroFonsecaDEV**
So I made some improvements. 
Your Bug:
The bug was because you were printing on the console the error message from your fetch() instead of considering that error as a BAD URL (if fetch fails because the URL is BAD). 

-Request only HEADERS. (optimization)
Now your fetch just requests the HEADERS. It will save time and space for your application.

-Timeout of 1.5secs (optimization)
If a server does not send a life-signal in 1.5secs, your fetch will throw an error. 

-Callback hell.
You were in callback hell. A lot of "IFs" nested and so on. (David has a good video about it. I will send the link to you).
I tried to change it, but you can still working on it. 

-function messageOne()
I just put your message inside a function to help us exit from callback hell.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
     backgroundColor: "#FFFFFF"
 }
 
-
-
-const version = boxen(chalk.redBright.bold("Urltester 1.0.0"), boxenOptions);
-
-if(process.argv.length == 2 ){
-    
+const messageOne = () => {
```
Comment (PedroFonsecaDEV) [inline index.js:17]: function to avoid help us avoid callback hell.

File: index.js [hunk 2]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
...
     console.log(chalk.red("\n                Standard user manual                "));
     console.log((chalk.gray("----------------------------------------------------"))); 
     console.log(chalk.redBright("urltester filename   -"), "reports good,bad,unknown urls"); 
     console.log(chalk.redBright("urltester v |version -"), "displays tool version"); 
     console.log((chalk.redBright("----------------------------------------------------\n"))); 
-}
+};
```
Comment (PedroFonsecaDEV) [inline index.js:23]: function to avoid help us avoid callback hell.

File: index.js [hunk 3]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
...
-        console.log(version);
-    }
-    else {      
-        const filePath = path.join(__dirname,process.argv[2]);
-        fs.readFile(filePath,'utf-8', function(err, data) {
-            if(err) {
-                console.log(chalk.red("Unsuccesful to read file"), err)
-            }
-            else {
-                const urlArr = data.match(/(http|https)(:\/\/)([\w+\-&@`~#$%^*.=/?:]+)/gi);
```
Comment (PedroFonsecaDEV) [inline index.js:41]: callback hell

File: index.js [hunk 4]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
...
-                        }
-                        else {
-                            console.log(chalk.gray(response.status, url) );
-                        }
-                    })
-                    .catch(function(error) {
-                        console.log(chalk.redBright("Error occured "), error);  
-                    });
-                });
-            }
```
Comment (PedroFonsecaDEV) [inline index.js:61]: callback hell

File: index.js (hunk 3 above)
Comment (PedroFonsecaDEV) [inline index.js:41]: callback hell

File: index.js [hunk 5]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
...
-        fs.readFile(filePath,'utf-8', function(err, data) {
-            if(err) {
-                console.log(chalk.red("Unsuccesful to read file"), err)
-            }
-            else {
-                const urlArr = data.match(/(http|https)(:\/\/)([\w+\-&@`~#$%^*.=/?:]+)/gi);
+const version = boxen(chalk.redBright.bold("Urltester 1.0.0"), boxenOptions);
+
+if(process.argv.length == 2 ) messageOne();
+else if(process.argv[2] === "v" || process.argv[2] === "version" || process.argv[2] === "-v") {
```
Comment (PedroFonsecaDEV) [inline index.js:28]: I also add the -v. 
And I suggest you to use only --v and/or -v. 
But it's just a suggestion.

File: index.js [hunk 6]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
...
-                    .catch(function(error) {
-                        console.log(chalk.redBright("Error occured "), error);  
-                    });
-                });
-            }
+        urlArr.forEach((url) => {
+            //network request of url
+            fetch(url,{method: "HEAD", timeout: 1500})
+            .then((response) => {
+            if(response.status == 200) console.log(chalk.green(response.status, url));
```
Comment (PedroFonsecaDEV) [inline index.js:42]: If your "IF" just execute one call you don't need to write: 
if(true){
console.log("hello")
} 
You can just write:
if(true) console.log("hello");

File: index.js [hunk 7]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
...
-                            console.log(chalk.green(response.status, url));
-                        }
-                        else if(response.status == 400 || response.status == 404) {
-                            console.log(chalk.red(response.status, url));
-                        }
-                        else {
-                            console.log(chalk.gray(response.status, url) );
-                        }
-                    })
-                    .catch(function(error) {
```
Comment (PedroFonsecaDEV) [inline index.js:57]: You can start to use arrow functions instead of the keyword function(){};

File: index.js [hunk 8]
```diff
@@ -14,51 +14,42 @@ const boxenOptions = {
...
+            //network request of url
+            fetch(url,{method: "HEAD", timeout: 1500})
+            .then((response) => {
+            if(response.status == 200) console.log(chalk.green(response.status, url));
+                        
+            else if(response.status == 400 || response.status == 404) console.log(chalk.red(response.status, url));
+        
+            else console.log(chalk.gray(response.status, url) );
+            })
+            .catch((error) => {
```
Comment (PedroFonsecaDEV) [inline index.js:48]: arrow function instead of function(){}

**Comment by PedroFonsecaDEV** [comment]
-Add feature:  request only headers.
-Add feature:  fetch timeout.
-Propose of "-v" instead of "v" 
-Better Formatting. (No callback hells)
-Probably I'm gonna create a new pull request with more improvements during this week.

Please let me know what do you think about these suggestions. 

Thanks.
Pedro

File: index.js (hunk 2 above)
Comment (mshinh) [reply index.js:23]: Thanks for this. It makes more sense now.

File: index.js (hunk 3 above)
Comment (mshinh) [reply index.js:41]: I'll read more into it.

File: index.js (hunk 5 above)
Comment (mshinh) [reply index.js:28]: Yeah, does no harm to add -v as well

File: index.js (hunk 8 above)
Comment (mshinh) [reply index.js:48]: Yeah arrow function is better

File: index.js (hunk 6 above)
Comment (mshinh) [reply index.js:42]: Will fix. thanks.

File: index.js (hunk 7 above)
Comment (mshinh) [reply index.js:57]: makes sense

**Review by mshinh — COMMENTED** [review]
The edits done make my code better and optimized.
Thanks for your help.

**Pull request merged**